File: BackEnd/BLL/Models/Ticket.cs
using SoftOS.Shared.Enums;

namespace SoftOS.BLL.Models
{
    public class Ticket
    {
        public int Id { get; set; }
        public int OrdemServicoId { get; set; }
        public int EmpresaId { get; set; }
        public int? ProfissionalId { get; set; }
        public int ClienteId { get; set; }
        public string Descricao { get; set; } = string.Empty;
        public DateTime DataCriacao { get; set; } = DateTime.Now;
        public DateTime? DataConclusao { get; set; }
        public bool Ativo { get; set; } = true;
        public TicketSituacao Situacao { get; set; } = TicketSituacao.Aberto;
        public virtual OrdemServico? OrdemServico { get; set; } = null;
        public virtual Empresa? Empresa { get; set; } = null;
        public virtual Profissional? Profissional { get; set; } = null;
        public virtual Cliente? Cliente { get; set; } = null;
        // Additional properties can be added as needed
    }
}

File: FrontEnd/src/typings/models/ticket.ts
import type { TicketSituacao } from 'enums';
import type { Cliente } from './cliente';
import type { Empresa } from './empresa';
import type { OrdemServico } from './ordemServico';
import type { Profissional } from './profissional';

export interface Ticket {
  id: number;
  ativo: boolean;
  clienteId: number;
  descricao: string;
  dataCriacao: string | number;
  dataConclusao: string | number;
  empresaId: number;
  ordemServicoId: number;
  profissionalId: number;
  situacao: TicketSituacao;

  readonly cliente?: Cliente[];
  readonly empresa?: Empresa[];
  readonly ordemServico?: OrdemServico[];
  readonly profissional?: Profissional[];
}
